internal/util/rocksmq/server/rocksmq/rocksmq_retention.go

```diff
@@ -278,39 +221,97 @@ func (ri *retentionInfo) expiredCleanUp(topic string) error {
 		}
 		curDeleteSize := deletedAckedSize + size
 		if msgSizeExpiredCheck(curDeleteSize, totalAckedSize) {
-			pageEndID, err = strconv.ParseInt(pKeyStr[FixedChannelNameLen+1:], 10, 64)
+			pageEndID, err = parsePageID(pKeyStr)
 			if err != nil {
 				return err
 			}
 			deletedAckedSize += size
+			pageCleaned++
 		} else {
 			break
 		}
 	}
+	if err := pageIter.Err(); err != nil {
+		return err
+	}
+
 	if pageEndID == 0 {
-		log.Debug("All messages are not expired")
+		log.Debug("All messages are not expired, skip retention", zap.Any("topic", topic))
 		return nil
 	}
-	log.Debug("ExpiredCleanUp: ", zap.Any("topic", topic), zap.Any("pageEndID", pageEndID), zap.Any("deletedAckedSize", deletedAckedSize))
+	log.Debug("Expired check by message size: ", zap.Any("topic", topic),
+		zap.Any("pageEndID", pageEndID), zap.Any("deletedAckedSize", deletedAckedSize), zap.Any("pageCleaned", pageCleaned))
+	return ri.cleanData(topic, pageEndID)
+}
+
+func (ri *retentionInfo) calculateTopicAckedSize(topic string) (int64, error) {
```

Comment:
Why here needs to do iteration again? Why not do the calculateTopicAckedSize with the ackedTs check? This need to load the same kv twice.

Reply:
Since I removed the persistent info of topicAckedSize, we will have to add up all page size through iteration to know the total size of all acked size.

I thought we have some optimization to iterate it only once, but since retention is not a critcal io path and pages should be limit, thus I just leave it there for further optimization

Reply:
OK